test/ess-inf-tests.el

```diff
@@ -49,12 +49,14 @@
   ;; (ess-async-command "{cat(1:5);Sys.sleep(5);cat(2:6)}\n" nil (get-process "R")
   ;;                    (lambda (proc) (message "done"))
   (with-r-running nil
-    (let ((output-buffer (get-buffer-create " *ess-async-text-command-output*"))
-          output)
-      (ess-async-command "{cat(1:5);Sys.sleep(0.5);cat(2:6)}\n" output-buffer (get-process "R")
-                         (lambda (proc string) (setq output string)))
-      (sleep-for 0.6)
-      (should (string= "2 3 4 5 6> " output)))))
+    (ess-async-command "{cat(1:5);Sys.sleep(0.5);cat(2:6)}\n"
+                       (get-buffer-create " *ess-async-text-command-output*")
+                       (get-process "R")
+                       (lambda (proc string)))
+    (loop repeat 3
+          until (null (process-get (get-process "R") 'callbacks))
+          do (sleep-for 0 600)
+          finally (should-not (process-get (get-process "R") 'callbacks)))))
```

Comment:
Does this need a `(should (process-get (get-process "R") 'callbacks))` before the loop? This way we test that a callback has indeed be registered, since you no longer check for side effects in the inferior.

You could also assign a flag variable like `called_back <- TRUE` in the callback and check for it after the loop, if you think that's more robust than checking for output. Perhaps doing both of these is best.

Reply:
> Does this need a `(should (process-get (get-process "R") 'callbacks))` before the loop?

Yes, it does.

Reply:
Thanks, these new checks are useful. As mentioned above, I would also like to check for the side effect in the inferior, please. It could be with a regexp search to make it more robust.

Reply:
Done.